File: CS/POCODependencies/PropertyChangedNotificationsUserControl.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Mvvm.POCO;
using DevExpress.Utils.MVVM;
using DevExpress.XtraEditors;
using System;
using System.Windows.Forms;

namespace DxSamplePOCODependencies {
    public partial class PropertyChangedNotificationsUserControl : XtraUserControl {
        public PropertyChangedNotificationsUserControl() {
            InitializeComponent();
            #region SetUp
            MVVMContext mvvmContext = new MVVMContext();
            mvvmContext.ContainerControl = this;
        
            TextEdit editor1 = new TextEdit();
            editor1.Dock = DockStyle.Top;
            editor1.Properties.Mask.MaskType = DevExpress.XtraEditors.Mask.MaskType.Numeric;
            editor1.Properties.Mask.EditMask = "n0";
            editor1.Properties.Mask.UseMaskAsDisplayFormat = true;
        
            TextEdit editor2 = new TextEdit();
            editor2.Dock = DockStyle.Top;
            editor2.Properties.Mask.MaskType = DevExpress.XtraEditors.Mask.MaskType.Numeric;
            editor2.Properties.Mask.EditMask = "n0";
            editor2.Properties.Mask.UseMaskAsDisplayFormat = true;
        
            LabelControl resultLabel = new LabelControl();
            resultLabel.Dock = DockStyle.Top;
            resultLabel.AutoSizeMode = LabelAutoSizeMode.Vertical;
        
            resultLabel.Parent = this;
            editor1.Parent = this;
            editor2.Parent = this;
            #endregion SetUp

            #region #propertyChangedNotifications
            // Set type of POCO-ViewModel
            mvvmContext.ViewModelType = typeof(SumViewModel);
            // Data binding for the operands
            mvvmContext.SetBinding(editor1, e => e.EditValue, "Operand1");
            mvvmContext.SetBinding(editor2, e => e.EditValue, "Operand2");
            // Data binding for the result
            mvvmContext.SetBinding(resultLabel, l => l.Text, "ResultText");
            #endregion #propertyChangedNotifications
        }
        #region CleanUp
        void OnDisposing() {
            var context = MVVMContext.FromControl(this);
            if(context != null) context.Dispose();
        }
        #endregion CleanUp
        // 
        // POCO View Model supports attributes and extension-methods for different purposes.
        // 
        public class SumViewModel {
            public SumViewModel() {
                Operand1 = 2;
                Operand2 = 2;
            }
            // using the BindableProperty attribute
            [BindableProperty(OnPropertyChangedMethodName = "NotifyResultAndResultTextChanged")]
            public virtual int Operand1 { get; set; }
            // using the BindableProperty attribute
            [BindableProperty(OnPropertyChangedMethodName = "NotifyResultAndResultTextChanged")]
            public virtual int Operand2 { get; set; }
            // We will raise change-notification for this property manually
            public int Result {
                get { return Operand1 + Operand2; }
            }
            // We will raise change-notification for this property manually
            public string ResultText {
                get { return string.Format("The result of operands summarization is: {0:n0}", Result); }
            }
            protected void NotifyResultAndResultTextChanged() {
                this.RaisePropertyChanged(x => x.Result); // change-notification for the Result
                this.RaisePropertyChanged(x => x.ResultText); // change-notification for the ResultText
            }
        }
    }
}

File: VB/POCODependencies/PropertyChangedNotificationsUserControl.vb
Imports DevExpress.MVVM
Imports DevExpress.MVVM.DataAnnotations
Imports DevExpress.MVVM.POCO
Imports DevExpress.Utils.MVVM
Imports DevExpress.XtraEditors
Imports System
Imports System.Windows.Forms

Namespace DxSamplePOCODependencies
    Partial Public Class PropertyChangedNotificationsUserControl
        Inherits DevExpress.XtraEditors.XtraUserControl
        Public Sub New() 
            InitializeComponent()
            Dim mvvmContext As New MVVMContext()
            mvvmContext.ContainerControl = Me
        
            Dim editor1 As New TextEdit()
            editor1.Dock = DockStyle.Top
            editor1.Properties.Mask.MaskType = DevExpress.XtraEditors.Mask.MaskType.Numeric
            editor1.Properties.Mask.EditMask = "n0"
            editor1.Properties.Mask.UseMaskAsDisplayFormat = True
        
            Dim editor2 As New TextEdit()
            editor2.Dock = DockStyle.Top
            editor2.Properties.Mask.MaskType = DevExpress.XtraEditors.Mask.MaskType.Numeric
            editor2.Properties.Mask.EditMask = "n0"
            editor2.Properties.Mask.UseMaskAsDisplayFormat = True
        
            Dim resultLabel As New LabelControl()
            resultLabel.Dock = DockStyle.Top
            resultLabel.AutoSizeMode = LabelAutoSizeMode.Vertical
        
            resultLabel.Parent = Me
            editor1.Parent = Me
            editor2.Parent = Me

            ' Set type of POCO-ViewModel
            mvvmContext.ViewModelType = GetType(SumViewModel)
            ' Data binding for the operands
            mvvmContext.SetBinding(editor1, Function(e) e.EditValue, "Operand1")
            mvvmContext.SetBinding(editor2, Function(e) e.EditValue, "Operand2")
            ' Data binding for the result
            mvvmContext.SetBinding(resultLabel, Function(l) l.Text, "ResultText")
        End Sub
         Private Sub OnDisposing()
            Dim context = MVVMContext.FromControl(Me)
            If context IsNot Nothing Then
                context.Dispose()
            End If
        End Sub
        '' 
        '' POCO View Model supports attributes and extension-methods for different purposes.
        '' 
        Public Class SumViewModel
            Public Sub New()
                Operand1 = 2
                Operand2 = 2
            End Sub
            ' using the BindableProperty attribute
            <BindableProperty(OnPropertyChangedMethodName:="NotifyResultAndResultTextChanged")>
            Public Overridable Property Operand1() As Integer
            ' using the BindableProperty attribute
            <BindableProperty(OnPropertyChangedMethodName:="NotifyResultAndResultTextChanged")>
            Public Overridable Property Operand2() As Integer
            ' We will raise change-notification for this property manually
            Public ReadOnly Property Result() As Integer
                Get
                    Return Operand1 + Operand2
                End Get
            End Property
            ' We will raise change-notification for this property manually
            Public ReadOnly Property ResultText() As String
                Get
                    Return String.Format("The result of operands summarization is: {0:n0}", Result)
                End Get
            End Property
            Protected Sub NotifyResultAndResultTextChanged()
                Me.RaisePropertyChanged(Function(x) x.Result) ' change-notification for the Result
                Me.RaisePropertyChanged(Function(x) x.ResultText) ' change-notification for the ResultText
            End Sub
        End Class
    	End Class
End Namespace